domain/state.go

```diff
@@ -95,3 +97,65 @@ func (st *StateBase) Prepare(query string, typeSamples ...any) (*sqlair.Statemen
 	st.statements[query] = stmt
 	return stmt, nil
 }
+
+type processFunc = func(in, out, samples []any) ([]any, []any, []any)
+
+func In[T any](v T) processFunc {
+	return func(in, out, samples []any) ([]any, []any, []any) {
+		return append(in, v), out, append(samples, v)
+	}
+}
+
+func Out[T any](v *T) processFunc {
+	return func(in, out, samples []any) ([]any, []any, []any) {
+		return in, append(out, v), append(samples, *v)
+	}
+}
+
+func MultiOut[T any](v *[]T) processFunc {
+	return func(in, out, samples []any) ([]any, []any, []any) {
+		var val T
+		return in, append(out, v), append(samples, val)
+	}
+}
+
+func (st *StateBase) QueryRow(ctx context.Context, tx *sqlair.TX, query string, args ...processFunc) error {
+	s, in, out, err := st.prepareWithArgs(query, args)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	return errors.Trace(tx.Query(ctx, s, in...).Get(out...))
+}
+
+func (st *StateBase) Query(ctx context.Context, tx *sqlair.TX, query string, args ...processFunc) error {
```

Comment:
This brings us closer to the std txn ergonomics which is nice.

Reply:
Yeah, `Query`, `QueryRow`, `Exec`.

Reply:
I'm worried that adding these function names instead of using the SQLair pattern of Get/GetAll/Run will mean that we will never move back to the SQLair methods even when some sort of convenience methods are added to it, or other improvements are made to the design. If domain switches to using these functions then we are effectively sidelining SQLair.

Reply:
It would be possible to achieve the same convenience with a method `st.Query(ctx, tx, queryString, args...)` that also removed the need for prepare but didn't replace the interface.